fdbclient/Schemas.cpp

```diff
@@ -1018,6 +1018,10 @@ const KeyRef JSONSchemas::statusSchema = R"statusSchema(
          "data_cluster_id" : 12346,
          "num_data_clusters":10
       },
+      "encryption_at_rest": {
+         "kms_is_healthy": true,
```

Comment:
Do we consider to have an intermediate state (like "flaky") between healthy/unhealthy. The currently code just take the result of the latest KMS request as indication of KMS being healthy or unhealthy. In reality the KMS can be in a flaky status where it only occasionally fail, and don't bring FDB completely down. The flaky state would be useful to be shown in status json to reveal KMS issues, even when FDB is running fine in the state (but probably degraded).

Reply:
I think flakiness is important to consider if our polling interval is shorter. Currently we poll every 30s. It's unlikely that a connection is flaky for 30s seconds (most likely its down or its up) (most other components also use a binary system to indicate health). I think we can consider adding a flaky status in a separate pr but would need a bit more thought to consider what is considered flaky (i.e maybe we take an average of the last N statuses).

Reply:
Eventually when we add metrics we should be able to deduce by looking at the charts whether a connection is flaky (i.e keeps switching from healthy to unhealthy). Not sure if it's super urgent right now

Reply:
If I didn't misread it, it is polling for the status (success or not) of the latest KMS request. The request might just get bad luck say getting a timeout against a healthy KMS, or it could also be the inverse. So I feel the status might not be very indicative of the healthiness of the KMS. Regarding the definition of flakiness, I'm thinking we can, say, keep history of successful/failure KMS request count in say last 5 seconds and see if the most of the requests success.

But I agree with you that the more fine-grained status of KMS would be easier to see in dashboard.

Reply:
@sfc-gh-nwijetunga is there a reason `KMS` health is under EaR banner; that doesn't seems right; especially because we allow KMS communication available without EaR database config being set. 

I think this is what we discussed offline, `kms health status` needs to be its own state which will impact EaR status section, but will NOT be contained in it.
Thanks

Reply:
Also, today EKP is the way to access KMS (which can disentangle) in future, but it is OK to keep EKP health into it, and yes it might change in future.

Reply:
> @sfc-gh-nwijetunga is there a reason KMS health is under EaR banner; that doesn't seems right; especially because we allow KMS communication available without EaR database config being set.

Sure I can move the KMS to a section outside `encryption_at_rest`